captcha-plugin-draft.txt

```diff
@@ -107,6 +107,13 @@ We provisionally instantiate these with
 
 [TODO: I would really like to use something other than RSA, but haven't found
 satisfactory research on alternative blind signature schemes.]
+[Alex: It could be possible to use a DSA-style blind signature (see
+http://blog.cryptographyengineering.com/p/note-on-blind-signature-schemes.html).
+The protocol requires three steps, the additional step is required since the
+server initially sends a random element for each signature that needs to be
+blindly signed. While this is not ideal it may be possible to incorporate
+this into the structure below so that the plugin asks for 10000 elements of
+randomness from the server when it generates the nonces.]
```

Comment:
Another option might be Boneh–Lynn–Shacham (BLS) signatures [1,2] which do not need the additional round trip of Schnorr signatures. They do need pairing-based crypto though. Blind BLS signatures [3,p12ff] can be realised similarly efficient as RSA. There is even a library for BLS signatures [4] but I don't know if it is ready to be used in the field.

[1] https://en.wikipedia.org/wiki/Boneh%E2%80%93Lynn%E2%80%93Shacham
[2] http://crypto.stanford.edu/~dabo/papers/weilsigs.ps
[3] https://eprint.iacr.org/2002/118.pdf
[4] https://crypto.stanford.edu/pbc/